src/InstrumentationEngine.ProfilerProxy/dllmain.cpp

```diff
@@ -108,12 +109,26 @@ namespace ProfilerProxy
         HRESULT hr = S_OK;
 
         *pHasProfiler = FALSE;
+        DWORD dError = 0;
 
         WCHAR wszEngineFullPath[MAX_PATH];
         IfFailRetErrno_Proxy(wcscpy_s(wszEngineFullPath, MAX_PATH, wszProfilerPath));
 
-        IfFailRet_Proxy(PathCchAppend(wszEngineFullPath, MAX_PATH, wszVersionFolder));
-        IfFailRet_Proxy(PathCchAppend(wszEngineFullPath, MAX_PATH, profilerRelativeFileName));
+        // For Windows 7 support, we cannot use PathCch functions.
+        //IfFailRet_Proxy(PathCchAppend(wszEngineFullPath, MAX_PATH, wszVersionFolder));
+        //IfFailRet_Proxy(PathCchAppend(wszEngineFullPath, MAX_PATH, profilerRelativeFileName));
+        WCHAR wszBuffer[2 * MAX_PATH]; // set to 2x the MAX_PATH in order to prevent buffer overflow.
+        if (!PathCanonicalize(wszBuffer, wszEngineFullPath) ||
+            !PathAppend(wszBuffer, wszVersionFolder) ||
+            !PathAppend(wszBuffer, profilerRelativeFileName) ||
+            wcslen(wszBuffer) >= MAX_PATH)
+        {
+            dError = GetLastError();
+            eventLogger.LogError(_T("dllmain::HasProfilerDll - Unable to generate fullpath to CLRIE dll: '%s' + '%s' + '%s'"), wszEngineFullPath, wszVersionFolder, profilerRelativeFileName);
+            return dError == 0 ? E_FAIL : HRESULT_FROM_WIN32(dError);
+        }
```

Comment:
Looks like duplicated logic. Can it be pulled into Common.Lib?

Reply:
refactoring made these functions slightly different